static/js/topic_list_data.js

```diff
@@ -33,14 +33,8 @@ export function get_list_info(stream_id, zoomed) {
         const num_unread = unread.num_unread_for_topic(stream_id, topic_name);
         const is_active_topic = active_topic === topic_name.toLowerCase();
         const is_topic_muted = muted_topics.is_topic_muted(stream_id, topic_name);
-        const resolved = topic_name.startsWith(resolved_topic.RESOLVED_TOPIC_PREFIX);
-        let topic_display_name = topic_name;
-
-        if (resolved) {
-            topic_display_name = topic_display_name.slice(
-                resolved_topic.RESOLVED_TOPIC_PREFIX.length,
-            );
-        }
+        const resolved = resolved_topic.is_resolved(topic_name);
+        const topic_display_name = resolved_topic.unresolve_name(topic_name);
```

Comment:
This block is the only product change proposed in the PR. 

Essentially, the question is how should we display topics that start with unlikely prefixes like `"✔ ✔✔ topic name`. The present implementation does the following:
* If you click the resolve toggle, just strips them all and gives you `topic name`.
* However, we display in the left sidebar the `✔✔ topic name` part after the initial `✔ `. We chose this design because otherwise one can have the sidebar show several topics that all appear to be a resolved `topic name` topic, even though the raw topics are totally different, which seems poor as a situation to be possible.

I think it's not particularly important in that this class of topic is a crazy/weird thing to do, but I think this design is probably worth preserving and documenting. I think what that means is we need another function name for this use case; perhaps `resolved_topic.display_name(topic_name)`.

The other more minor coding detail is I think `resolved_topic.unresolve_name` cannot be called when `resolved` isn't known to be true, since otherwise one would strip the prefix from `✔topic name` (note no space), which is not recognized as a resolved topic. I would consider addressing this by having the `display_name` function including the `is_resolved` check, and return the topic unmodified if not resolved.  (One could consider having it be a function that returns two things, I guess, but I'm not convinced that'll feel cleaner).

Reply:
> Essentially, the question is how should we display topics that start with unlikely prefixes like `"✔ ✔✔ topic name`. The present implementation does the following:

I see. Sure, that reasoning makes sense. Definitely calls for a comment explaining it, which I can add.



> The other more minor coding detail is I think `resolved_topic.unresolve_name` cannot be called when `resolved` isn't known to be true, since otherwise one would strip the prefix from `✔topic name` (note no space), which is not recognized as a resolved topic. I would consider addressing this by having the `display_name` function including the `is_resolved` check, and return the topic unmodified if not resolved.

I believe it already behaves that way (as its jsdoc says). That's why RESOLVED_TOPIC_PREFIX_RE starts with `^✔ ` and then has the *-repeated character class after that.

I'll add a test for that case to help make sure it doesn't regress in some future change.